Userland/Libraries/LibCore/FilePermissionsMask.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2021, Xavier Defrang <xavier.defrang@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Assertions.h>
+#include <AK/CharacterTypes.h>
+#include <AK/StringUtils.h>
+
+#include <LibCore/FilePermissionsMask.h>
+
+namespace Core {
+
+enum State {
+    Reference,
+    Mode
+};
+
+enum ClassFlag {
+    Other = 1,
+    Group = 2,
+    User = 4
+};
+
+enum Operation {
+    Add,
+    Remove,
+    Assign,
+};
+
+ErrorOr<FilePermissionsMask> FilePermissionsMask::parse(const StringView& string)
```

Comment:
Don't pass StringView by const reference, pass it by value (see https://github.com/SerenityOS/serenity/commit/8b1108e4858f797c9216dc8ae4a3918ad50c73b4)